lib/web/fetch/index.js

```diff
@@ -124,12 +125,68 @@ function handleFetchDone (response) {
   finalizeAndReportTiming(response, 'fetch')
 }
 
+// This will publish all diagnostic events only when we have subscribers.
+function ifSubscribersRunStores (req, input, init, callback) {
+  const hasSubscribers = subscribersCheck()
+
+  if (hasSubscribers) {
+    const context = { req, input, init, result: null, error: null }
+
+    return channels.start.runStores(context, () => {
+      try {
+        return callback(createInstrumentedDeferredPromise(context))
+      } catch (e) {
+        context.error = e
+        channels.error.publish(context)
+        throw e
+      } finally {
+        channels.end.publish(context)
+      }
+    })
+  } else {
+    return callback(createDeferredPromise())
+  }
+}
+
+// subscribersCheck will be called at the beginning of the fetch call
+// and will check if we have subscribers
+function subscribersCheck () {
```

Comment:
AFAIK `tracingChannel.hasSubscribers` will do all these checks for you. No need to do it manually. 

https://github.com/nodejs/node/blob/main/lib/diagnostics_channel.js#L280

cc: @Qard

Reply:
The `tracingChannel.hasSubscribers` getter is fairly new so this needs to continue doing things the old way for awhile.